### DISCUSSION TITLE
Access location.state in loader function

### DISCUSSION THREAD

**Discussion by RoBackman** [remix-run/remix#8243] [Q&A]
Hi!

Would anyone happen to know how you could access the react-router version of location.state inside a loader function
```
// The equivalent of this in a loader function
const { state } = useLocation()
```

**Comment by sergiodxa** [comment]
The location.state is a client-side only thing that the browser keeps in the History API, the only way to use it server-side would be to send it yourself to the server (on the request body for action or search params for loaders).

**Reply by RoBackman** [reply]
Seems I posted the question on the wrong repo, was meant for react-router...

But nonetheless it explains why it is not found there either (would break compatibility of loaders in case of migrating to the full remix experience)

And from what I tested `window.history.state` seems to be empty when accessing it in a loader. So guess it is query-params or post-body all the way then.
